### PR TITLE
add ability to vote on an article-task7

### PR DESCRIPTION
**Pull request by Kylehinks97**

### CODE REVIEW

File: fe-nc-news-project/src/Comments.jsx [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import { useEffect, useState } from "react";
 import { useParams } from "react-router-dom";
 import { fetchCommentsByArticleId } from "./utils";
+import Article from "./Article";
 
 export default function Comments() {
   const [comments, setComments] = useState([]);
```
Comment (HannahBurks) [inline fe-nc-news-project/src/Comments.jsx:7]: We also want to have a loading state for our comments

File: fe-nc-news-project/src/utils.js [hunk 2]
```diff
@@ -23,12 +22,32 @@ export function fetchArticlesById(article_id) {
         console.log(err, "request error");
     })
 }
-
 export function fetchCommentsByArticleId(article_id) {
     return api.get(`/articles/${article_id}/comments`)
     .then((result) => {
         return result
     }).catch((err) => {
         console.log(err, "request error");
```
Comment (HannahBurks) [inline fe-nc-news-project/src/utils.js]: We don't need .catch in our api file. We want to add a .catch when we invoke the util function in the components.

File: fe-nc-news-project/src/utils.js [hunk 3]
```diff
@@ -23,12 +22,32 @@ export function fetchArticlesById(article_id) {
...
+    }).catch((err) => {
+      console.log("failed to upvote");
+    })
+  } else if (direction === "down") {
+    const downvoteObj = {}
+    downvoteObj["inc_votes"] = -1
+    return api.patch(`/articles/${article_id}`, downvoteObj)
+    .then((result) => {
+      return result
+    }).catch((err) => {
```
Comment (HannahBurks) [inline fe-nc-news-project/src/utils.js]: Make sure to remove any .catch on this file

**Review by HannahBurks — CHANGES_REQUESTED** [review]
Nice work Kyle, but a few changes ie loading state and also changing where our .catch() are :)

**Comment by HannahBurks** [comment]
I've also just tried the voting feature and on page refresh it doesn't work. Think about why this may be

File: fe-nc-news-project/src/utils.js [hunk 4]
```diff
@@ -5,30 +5,23 @@ const api = axios.create({
...
-    })
-}
\ No newline at end of file
+  return api.get(`/articles/${article_id}/comments`).then((result) => {
+    return result;
+  });
+}
+export function updateArticleVotes(obj, article_id) {
+  return api.patch(`/articles/${article_id}`, obj).then((result) => {
+    console.log(result.data.article.votes, "<-- in the util");
```
Comment (aloush-dev) [inline fe-nc-news-project/src/utils.js:24]: remove please

**Review by aloush-dev — APPROVED** [review]
nice work.
feel free to merge

**Pull request merged**